chunky/src/java/se/llbit/chunky/block/DecoratedPot.java

```diff
@@ -0,0 +1,69 @@
+package se.llbit.chunky.block;
+
+import se.llbit.chunky.model.DecoratedPotModel;
+import se.llbit.chunky.resources.Texture;
+import se.llbit.chunky.entity.Entity;
+import se.llbit.math.Vector3;
+import se.llbit.nbt.CompoundTag;
+import se.llbit.nbt.StringTag;
+import se.llbit.nbt.Tag;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class DecoratedPot extends AbstractModelBlock {
+
+  private final String facing;
+  private final String description;
+
+  public DecoratedPot(String facing, boolean waterlogged, String[] shards) {
+    super("decorated_pot", Texture.decoratedPotSide);
+    this.waterlogged = waterlogged;
+    this.facing = facing;
+    description = "waterlogged=" + waterlogged
+      + ", facing=" + facing
+      + ", shards=" + Arrays.stream(shards)
+      .map(shard -> shard == null ? "minecraft:brick" : shard)
+      .collect(Collectors.joining(", ", "[", "]"));
+    model = new DecoratedPotModel(facing, shards);
+  }
+
+  @Override
+  public String description() {
+    return description;
+  }
+
+  @Override
+  public boolean isBlockEntity() {
+    return true;
+  }
+
+  @Override
+  public boolean isModifiedByBlockEntity() {
+    return true;
+  }
+
+  @Override
+  public Entity toBlockEntity(Vector3 position, CompoundTag entityTag) {
+    return new DecoratedPotModel.DecoratedPotSpoutEntity(position, facing);
+  }
+
+  @Override
+  public Tag getNewTagWithBlockEntity(Tag blockTag, CompoundTag entityTag) {
+    CompoundTag newBlockTag = new CompoundTag();
```

Comment:
This could be used to later render this as a block. I'd rename the "special" tag to prefix it with something unique to chunky, similar to water, lava of legacy blocks.

Reply:
Sounds reasonable - let's discuss the changes on the main repo PR :)